jsonslice_test.go

```diff
@@ -336,6 +336,37 @@ func Test_Expressions(t *testing.T) {
 	}
 }
 
+func Test_FuncNow(t *testing.T) {
+
+	tests := []struct {
+		name  string
+		Query string
+		isNil bool
+	}{
+		{"happy path", "$.now()", false},
+		{"invalid use of now function, will return nil", "$.x.now()", true},
+		{"invalid use of now function, will return nil root level", "$.now(\"2006-01-02T15:04:05.000Z\")", true},
```

Comment:
It could be useful to support users passing in a custom format, to allow flexibility for users.

$.now() → “2023-04-05T16:15:11.476084178Z" // default ISO8601

$.now(RFC3339) → “2023-04-05T16:15:11Z47:60" // shorthand support

$.now("2006-01-02T15:04:05Z07:00") → “2023-04-05T16:15:11Z47:60"

Reply:
I looked into this, seems like adding ability for function may open up a can of worm. by default json functions don't take param with (). usually the param it takes is the node where the traversal is happening. Introducing a different pattern of param taking is anti pattern and could cause unintentional behaviour.

what if instead of taking param, we can add more func. eg
$.now
$.nowRFC3339
and so on and so forth.

Reply:
Ah I see. I like that idea. Maybe the syntax could be `$.now().RFC3339()`?

Alternatively, formatting could be a transform.